src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php

```diff
@@ -316,6 +316,17 @@ protected function compileSelected($condition)
         return "<?php if{$condition}: echo 'selected'; endif; ?>";
     }
 
+    /**
+     * Compile a bool block into valid PHP.
+     *
+     * @param  bool  $condition
+     * @return string
+     */
+    protected function compileBool($condition)
+    {
+        return "<?php if{$condition}: echo 'true'; else: 'false'; endif; ?>";
```

Comment:
Your code doesn't print anything for the "false" path.

return  "<?php echo {$condition} ? 'true' : 'false'; ?>";